### PR TITLE
Handle return value 24 the same was as 23.

### PR DESCRIPTION
**Pull request by merks**
https://github.com/eclipse-equinox/p2/issues/166

### CODE REVIEW

**Comment by merks** [comment]
@umairsair

I'm not sure how to test if this is needed but it wold seem consistent with the exiting handling for 23.

File: bundles/org.eclipse.equinox.launcher/src/org/eclipse/equinox/launcher/Main.java [hunk 1]
```diff
@@ -1438,7 +1438,7 @@ public static void main(String[] args) {
 			// If the return code is 23, that means that Equinox requested a restart.
 			// In order to distinguish the request for a restart, do a System.exit(23)
 			// no matter of 'osgi.noShutdown' runtime property value.
-			if (!Boolean.getBoolean(PROP_NOSHUTDOWN) || result == 23)
+			if (!Boolean.getBoolean(PROP_NOSHUTDOWN) || result == 23 || result == 24)
```
Comment (iloveeclipse) [inline bundles/org.eclipse.equinox.launcher/src/org/eclipse/equinox/launcher/Main.java:1441]: Ed, the comment has to be updated

Comment (merks) [reply bundles/org.eclipse.equinox.launcher/src/org/eclipse/equinox/launcher/Main.java:1441]: You think?  😱 😖

Comment (iloveeclipse) [reply bundles/org.eclipse.equinox.launcher/src/org/eclipse/equinox/launcher/Main.java:1441]: Well, comment says something about 23, but then we have 23 and 24, with no idea why 24 is needed

Comment (merks) [reply bundles/org.eclipse.equinox.launcher/src/org/eclipse/equinox/launcher/Main.java:1441]: I was only joking.   If you read the linked issue you will know why 24 is needed.  In any case, note how the comment is actually wrong and or simply misleading already to suggest that it does System.exit(23) when the code actually does System.exit(result).  I'll fix it.

**Comment by laeubi** [comment]
I just wanted to link this one here regarding exit codes it sometimes comes to a surprise:
- https://github.com/eclipse-equinox/equinox/discussions/378

**Comment by merks** [comment]
This code has not been touched in a very long time so I doubt there is any one around who can tell us what's right or wrong.  I think consistently handing 24 as we do 23 is most likely the correct and consistent thing to do and it's what I've done elsewhere to fix actual problems, e.g.,

https://github.com/eclipse-pde/eclipse.pde/pull/1121

**Comment by HannesWell** [comment]
I wonder if there should be some global constants for that magic numbers?

**Comment by laeubi** [comment]
> I wonder if there should be some global constants for that magic numbers?

You mean like https://github.com/eclipse-equinox/equinox/blob/850923a5273e11487e5a0b2b505bfdef637d7d2d/bundles/org.eclipse.equinox.app/src/org/eclipse/equinox/app/IApplication.java#L48 :-)

**Comment by HannesWell** [comment]
> > I wonder if there should be some global constants for that magic numbers?
> 
> You mean like

For example, yes. :)
I think it would be good to use that where possible, i.e. platform and PDE. Here it is probably not accessible.

**Comment by merks** [comment]
I cancel this because https://github.com/eclipse-platform/eclipse.platform.ui/pull/1622 avoids surfacing that exit code.

**Pull request closed without merging**